raiden/network/transport/matrix.py

```diff
@@ -662,58 +737,58 @@ def _get_room_for_address(
         )
         return room
 
-    def _get_unlisted_room(self, room_name, invitees):
+    def _get_unlisted_room(self, room_name, invitees: List[User]):
         """Obtain a room that cannot be found by search_room_directory."""
         room_name_full = f'#{room_name}:{self._server_name}'
-        room_not_found = False
+        invitees_uids = [user.user_id for user in invitees]
 
-        for _ in range(10):
-            if room_not_found:
-                try:
-                    room = self._client.create_room(
-                        room_name,
-                        invitees=[user.user_id for user in invitees],
-                        is_public=True,  # FIXME: debug only
+        for _ in range(5):
+            # try joining room
+            try:
+                room = self._client.join_room(room_name_full)
+            except MatrixRequestError as error:
+                if error.code == 404:
+                    self.log.info(
+                        f'Room {room_name_full} not found, trying to create it.',
+                        error=error,
                     )
-                except MatrixRequestError as error:
-                    if error.code == 409:
-                        message = 'seems to have been created by peer meanwhile.'
-                    else:
-                        message = f'{error.code} {error.content}'
-                    self.log.info(f'Error creating room {room_name}: {message}. '
-                                  f'Retrying to join...')
-                    room_not_found = False
                 else:
-                    self.log.info('Room created successfully', room=room, invitees=invitees)
-                    return room
+                    self.log.info(f'Error joining room {room_name}: '
+                                  f'{error.content} {error.code}')
             else:
-                try:
-                    room = self._client.join_room(room_name_full)
-                except MatrixRequestError as error:
-                    if error.code == 404:
-                        self.log.info(
-                            f'Room {room_name_full} not found, trying to create it.',
-                            error=error,
-                        )
-                        room_not_found = True
-                    else:
-                        self.log.info(f'Error joining room {room_name}: '
-                                      f'{error.content} {error.code}')
+                self.log.info('Room joined successfully', room=room)
+                break
+
+            # if can't, try creating it
+            try:
+                room = self._client.create_room(
+                    room_name,
+                    invitees=invitees_uids,
+                    is_public=True,  # FIXME: debug only
```

Comment:
Please, open an issue for this. It could also include the encryption required by #473

Reply:
Done: #2302